import-export-cli/integration/api_test.go

```diff
@@ -63,3 +71,362 @@ func TestExportImportApiAdminSuperTenantUser(t *testing.T) {
+		apiProvider: credentials{username: superTenantApiCreator, password: superTenantApiCreatorPassword},
+		ctlUser:     credentials{username: superTenantAdminUsername, password: superTenantAdminPassword},
+		api:         api,
+		srcAPIM:     dev,
+		destAPIM:    prod,
+	}
+
+	validateAPIExport(t, args)
+
+	// Since --preserve-provider=false is not specified, the apiProvider remain as it is and the ctlUser is tenant admin
+	args.ctlUser = credentials{username: tenantAdminUsername, password: tenantAdminPassword}
+
+	// Import the API to env2 as tenant admin across domains
+	validateAPIImportFailure(t, args)
+}
+
+func TestListApisAdminSuperTenantUser(t *testing.T) {
+	adminUsername := superAdminUser
+	adminPassword := superAdminPassword
+
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add a random number (< 10) of API (same copy)
+	randomNumber := rand.Intn(10)
+	for apiCount := 0; apiCount <= randomNumber; apiCount++ {
+		// Add the API to env1
+		addAPI(t, dev, apiCreator, apiCreatorPassword)
+	}
+
+	args := &apiImportExportTestArgs{
+		ctlUser: credentials{username: adminUsername, password: adminPassword},
+		srcAPIM: dev,
+	}
+
+	validateAPIsList(t, args)
+}
+
+func TestListApisAdminTenantUser(t *testing.T) {
+	tenantAdminUsername := superAdminUser + "@" + TENANT1
+	tenantAdminPassword := superAdminPassword
+
+	apiCreator := creator.UserName + "@" + TENANT1
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add a random number (< 10) of API (same copy)
+	randomNumber := rand.Intn(10)
+	for apiCount := 0; apiCount <= randomNumber; apiCount++ {
+		// Add the API to env1
+		addAPI(t, dev, apiCreator, apiCreatorPassword)
+	}
+
+	args := &apiImportExportTestArgs{
+		ctlUser: credentials{username: tenantAdminUsername, password: tenantAdminPassword},
+		srcAPIM: dev,
+	}
+
+	validateAPIsList(t, args)
+}
+
+func TestDeleteApiAdminSuperTenantUser(t *testing.T) {
+	adminUsername := superAdminUser
+	adminPassword := superAdminPassword
+
+	apiCreator := creator.UserName
+	apiCreatorPassword := creator.Password
+
+	dev := apimClients[0]
+
+	// Add a random number (< 10) of APIs (same copy)
+	randomNumber := rand.Intn(10)
+
+	var api *apim.API
+	for apiCount := 0; apiCount <= randomNumber; apiCount++ {
+		// Add the API to env1 (Since we are deleting an API later, if the auto clean is enabled an error may occur when
+		// trying to delete the already deleted API. So the auto cleaning will be disabled.)
+		api = addAPIWithoutCleaning(t, dev, apiCreator, apiCreatorPassword)
+	}
+
+	args := &apiImportExportTestArgs{
+		ctlUser: credentials{username: adminUsername, password: adminPassword},
+		api:     api, // Choose the last API from the loop to delete it
+		srcAPIM: dev,
+	}
+
+	t.Cleanup(func() {
+		apiList := getAPIs(dev, adminUsername, adminPassword)
+		for _, api := range apiList.List {
+			deleteAPI(t, dev, api.ID, adminUsername, adminPassword)
+		}
+	})
```

Comment:
We can simplify this pattern to reduce test complexity as follows,

1. Add some APIs in the loop by calling the `addAPI()` function.

2. Add a single API by calling the `addAPIWithoutCleaning()` function. This will be the actual API that will be deleted by the apictl in the test.

3. No need to register a special `t.Cleanup()` function within the test now. All APIs that were added using the `addAPI()` function will handle removing these APIs via the already builtin Cleanup function. apictl delete-api call will remove the API that is not automatically deleted. 

Test looks much simpler and cleaner now.

Reply:
That's true. Thank you for pointing this. Will do the changes where applicable including for API Products as well.